tools/spatialDraw/DrawingManager.js

```diff
@@ -0,0 +1,758 @@
+        this.triggerCallbacks('tool', tool);
+    }
+
+    /**
+     * Sets the cursor type.
+     * @param {DrawingManager.Cursor} cursor - The cursor type.
+     */
+    setCursor(cursor) {
+        this.setEraseMode(false);
+        this.tool.endDraw(this.drawingGroup, this.cursor.getPosition());
+        this.cursor = cursor;
+        this.triggerCallbacks('cursor', cursor);
+    }
+
+    /**
+     * Sets the drawing color.
+     * @param {string} color - The color.
+     */
+    setColor(color) {
+        this.setEraseMode(false);
+        this.tool.endDraw(this.drawingGroup, this.cursor.getPosition());
+        this.tool.setColor(color);
+        this.triggerCallbacks('color', color);
+    }
+
+    /**
+     * Sets the drawing size.
+     * @param {number} size - The size.
+     */
+    setSize(size) {
+        this.tool.endDraw(this.drawingGroup, this.cursor.getPosition());
+        this.tool.setSize(size);
+        this.triggerCallbacks('size', size);
+    }
+
+    /**
+     * Enters or exits erase mode.
+     * @param {boolean} value - Value to set the erase mode to.
+     */
+    setEraseMode(value) {
+        this.erasing = value;
+        this.tool.endDraw(this.drawingGroup, this.cursor.getPosition());
+        this.triggerCallbacks('eraseMode', this.erasing);
+    }
+
+    /**
+     * Erases lines drawn under the pointer.
+     * @param {Object} pointerEvent - The triggering pointer event.
+     */
+    erase(pointerEvent) {
+        const position = {
+            x: (pointerEvent.pageX / window.innerWidth) * 2 - 1,
+            y: - (pointerEvent.pageY / window.innerHeight) * 2 + 1,
+        };
+
+        this.raycaster.setFromCamera(new THREE.Vector2(position.x, position.y), this.camera);
+        const serializedDrawings = [];
+        this.drawingGroup.children.forEach(obj => {
+            if (obj.serialized) {
+                serializedDrawings.push(obj);
+            }
+        });
+        const intersects = this.raycaster.intersectObjects(serializedDrawings);
+        intersects.forEach(intersect => {
+            if (intersect.object.serialized) {
+                const undoEvent = {
+                    type: 'draw',
+                    data: intersect.object.serialized
+                };
+                intersect.object.parent.remove(intersect.object);
+                this.pushEvent(undoEvent);
+            }
+        });
+    }
+
+    /**
+     * Calls updateCallbacks with the current state of the drawing.
+     */
+    shareUpdates() {
+        const serializedDrawings = this.serializeDrawing();
+        this.triggerCallbacks('update', serializedDrawings);
+    }
+
+    /**
+     * Adds an undoable event to the event stack.
+     * @param {Object} event - The event to be performed if undo is pressed.
+     */
+    pushEvent(event) {
+        this.eventStack.push(event);
+        this.shareUpdates();
+    }
+
+    /**
+     * Performs the most recent event in the event stack.
+     */
+    undoEvent() {
```

Comment:
This name doesn't really make sense to me if it's doing the latest non-erase event. Is this more like onUndoEvent/afterUndoEvent?

Reply:
Does popEvent make more sense, being the opposite of pushEvent? It does the latest event, regardless of whether or not it's a draw or erase event, so I'm not sure what you mean there.